### ISSUE TITLE
ReceiveMessages of ServiceBus SDK with DefaultCredentials gives error connlost

### ISSUE THREAD

**Issue by vidhyapg** [Azure/azure-sdk-for-go#20529]
-  sdk/messaging/azservicebus
- v1.2.1
- go version go1.20.1 linux/amd64

I am facing issue with ReceiveMessages in azservicebus sdk. I am getting error 
```(connlost): dial tcp: lookup bsrpdveubf23servicebus on 10.0.0.10:53: no such host```

using NewDefaultAzureCredential using AZURE_CLIENT_ID for MSI in cloud, like this.
``` credential, err := azidentity.NewDefaultAzureCredential(nil) ```

I could also repro issue locally where I am logged in az cli. I tried explicitly using NewAzureCLICredential, like this

``` credential, err := azidentity.NewAzureCLICredential(nil) ```
Still getting the same error.  

I know the credentials are correct because I can use the same credential and successfully create a Subscription for a Topic using [github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/servicebus/armservicebus](http://github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/servicebus/armservicebus) v1.0.0. 

I expect ReceiveMessages to work correctly and receive message using DefaultCrendetial(MSI and Az Cli). I can see it works correctly when I create a client with connectionString

```client, err = azservicebus.NewClientFromConnectionString(connectionString, nil)```

Can you please fix the bug? or provide more information on the error if it is something I can fix from client side?  

Thanks
Vidhya

**Comment by richardpark-msft** [comment]
Hey @vidhyapg, I think you might have omitted the suffix for the hostname when specifying the namespace name. The connection string always includes the full hostname, which is probably why it worked there.

You'll want to change `bsrpdveubf23servicebus` to  `bsrpdveubf23servicebus.servicebus.windows.net`, which is the fully qualified domain name. You can see this name in the Azure Portal view for your Service Bus under 'Host name'.

**Comment by vidhyapg** [comment]
Thank you @richardpark-msft. That indeed was the problem. It would have been better if the error was thrown earlier in azservicebus.NewClient. I did not expect ReceiveMessage to fail for this reason (error message was not super helpful) especially as I was successfully able to create a client with that namespace. One more thing to note if we are to be consistent is that CreateOrUpdate of armservicebus SubscriptionClient expects namespace to be without the qualified domain name. Anyways I am over that problem now. :)

**Comment by richardpark-msft** [comment]
Yeah, it's a consequence of our clients being lazy. We don't do any network IO until you use a function that has a context passed to it so the evaluation of your parameters can be later than expected.

The inconsistency is one that I've noted as well, and it's down to how the services view resources. In our case we're connecting to an actual host, so the suffix is required for DNS, etc. 

Closing, as I believe we've arrived at the answer.
